lib/LogCollector.js

```diff
@@ -0,0 +1,207 @@
+ * reaches or exceeds this limit.
+ * @param {number} options.storeInterval how often the logs should be stored in
+ * case <tt>maxEntryLength</tt> was not exceeded.
+ * @param {boolean} options.stringifyObjects indicates whether or not object
+ * arguments should be "stringified" with <tt>JSON.stringify</tt> when a log
+ * message is composed. Note that objects logged on the error log level are
+ * always stringified.
+ *
+ * @constructor
+ */
+function LogCollector(logStorage, options) {
+    this.logStorage = logStorage;
+    this.stringifyObjects = options ? options.stringifyObjects : false;
+    this.storeInterval = options ? options.storeInterval : 30000;
+    this.maxEntryLength = options ? options.maxEntryLength : 10000;
+    // Bind the log method for each level to the corresponding method name
+    // in order to implement "global log transport" object.
+    Object.keys(Logger.levels).forEach(
+    function (logLevel) {
+        const methodName = Logger.levels[logLevel];
+        this[methodName] = function (logLevel) {
+            this._log.apply(this, arguments);
+        }.bind(this, logLevel);
+    }.bind(this));
+    /**
+     * The ID of store logs interval if one is currently scheduled or
+     * <tt>null</tt> otherwise.
+     * @type {number|null}
+     */
+    this.storeLogsIntervalID = null;
+    /**
+     * The log messages that are to be batched into log entry when
+     * {@link LogCollector._flush} method is called.
+     * @type {string[]}
+     */
+    this.queue = [];
+    /**
+     * The total length of all messages currently stored in the {@link queue}.
+     * @type {number}
+     */
+    this.totalLen = 0;
+    /**
+     * The counter increased whenever next logs batch is stored (or cached if
+     * the storage is not ready yet).
+     * @type {number}
+     */
+    this.counter = 1;
+}
+
+/**
+ * The log method bound to each of the logging levels in order to implement
+ * "global log transport" object.
+ *
+ * @private
+ */
+LogCollector.prototype._log = function() {
+
+    var logLevel = arguments[0];
+    var msg = '';
+    for (var i = 1, len = arguments.length; i < len; i++) {
+        var arg = arguments[i];
+        // objects logged on error level are always converted to JSON
+        if ((this.stringifyObjects || logLevel === Logger.levels.ERROR)
+            && typeof arg === 'object') {
+            arg = stringify(arg);
+        }
+        msg += arg;
+        if (i != len - 1) {
+            msg += ' ';
+        }
+    }
+
+    if (msg.length) {
+        // FIXME add "same as the previous" message compression
+        this.queue.push(msg);
+        this.totalLen += msg.length;
+    }
+
+    if (this.totalLen >= this.maxEntryLength) {
+        this._flush(true /* reschedule */);
+    }
+};
+
+/**
+ * Starts this instance and schedules periodical "store logs" task.
+ */
+LogCollector.prototype.start = function () {
+    this._reschedulePublishInterval();
```

Comment:
@paweldomas Please discard this comment. I saw the code in jitsi-meet and you actually need this to cache the first logs, right?